workflow/executor/executor.go

```diff
@@ -40,7 +40,8 @@ import (
 	os_specific "github.com/argoproj/argo/workflow/executor/os-specific"
 )
 
-var MainContainerStartRetry = wait.Backoff{
+// ExecutorRetry is a retry backoff settings for WorkflowExecutor
+var ExecutorRetry = wait.Backoff{
 	Steps:    8,
```

Comment:
so the change here is 

- 8 steps at 1 second rather than 5 at 10ms

Reply:
Yes that's right. And this value was original from here: https://github.com/argoproj/argo/pull/3675

Reply:
![image](https://user-images.githubusercontent.com/1142830/91608690-e1896e80-e92a-11ea-96e2-62a4bb74c1df.png)
http://backoffcalculator.com/?attempts=8&rate=2&interval=1

Reply:
this seems wrong to me, wait a total of 4 minutes?

Reply:
![image](https://user-images.githubusercontent.com/1142830/91608797-0bdb2c00-e92b-11ea-93cf-42fa5e6508f5.png)
Try again - 2 vs 3 is BIG difference.

Reply:
Vs old one:


![image](https://user-images.githubusercontent.com/1142830/91608901-3dec8e00-e92b-11ea-8bf1-8e2314207736.png)

Reply:
@jessesuen @sarabala1979 this really make me think:

1. Our default backoff is not exponential at all. Should it be?
2. 10ms by default seems short to me. 

Are we creating a thundering herd problem?

What about different defaults?

1. interval = 20ms
2. factor = 2

![image](https://user-images.githubusercontent.com/1142830/91609272-f286af80-e92b-11ea-8f17-c7d3c9771d17.png)

Reply:
That is still <1s .

Reply:
I think this works or maybe update retry attempts to 6, then we will have one test case >1 s.
I feel it should cover most of cases ?

Reply:
I think we probably want to keep retries <1s because max duration for a reconciliation is 10s.

Reply:
I see, do you think whether we can replace all retries with this new default backoff, or we should still have a special backoff for executor, not sure whether the new default can work with the MainContainerStart @yoms ?

Reply:
I think the new default backoff will be too low for the executor, but I won't be able to verify it until a few weeks (holiday time...).
From my point of view, these values should be configurable as sometime Argo users have to tweak them depending on their hosting platform. 
Sorry for this useless answer :-)

Reply:
Agreed, make these values tweakable would be great! 
For now as a short-term patch, can we increase the timeout for executor only as what i did, then have a follow up to add the config for this, @alexec any suggestions ?